stores/users/store.go

```diff
@@ -0,0 +1,83 @@
+package users
+
+import (
+	"database/sql"
+	"errors"
+	"user-crud/models"
+	"user-crud/stores"
+)
+
+type userStore struct {
+	db *sql.DB
+}
+
+func New(db *sql.DB) stores.User {
+	return &userStore{db}
+}
+
+func (u *userStore) Insert(usr models.User) (models.User, error) {
+	query := "insert into user1 values(?,?,?,?,?)"
+	stmt, err := u.db.Prepare(query)
+	defer func() {
+		if err == nil {
+			stmt.Close()
+		}
+	}()
+	if err != nil {
+		return models.User{}, err
+	}
+	_, err = stmt.Exec(usr.Id, usr.Name, usr.Email, usr.Phone, usr.Age)
+	if err != nil {
+		return models.User{}, err
+	}
+	return usr, nil
+}
+
+func (u *userStore) GetAll() ([]models.User, error) {
+	var users []models.User
+	query := "select * from user1"
+	rows, err := u.db.Query(query)
+	defer func() {
+		if err == nil {
+			rows.Close()
+		}
+	}()
+
+	if err != nil {
+		return []models.User{}, err
+	}
+	for rows.Next() {
+		usr := models.User{}
+		rows.Scan(&usr.Id, &usr.Name, &usr.Email, &usr.Phone, &usr.Age)
+		users = append(users, usr)
+	}
+	println(users)
+	return users, nil
+}
+
+func (u *userStore) GetById(id int) (models.User, error) {
+	var usr models.User
+	query := "select * from user1 where id = ?"
+	rows := u.db.QueryRow(query, id)
+	err := rows.Scan(&usr.Id, &usr.Name, &usr.Email, &usr.Phone, &usr.Age)
+	return usr, err
+}
+
+func (u *userStore) Update(id int, name string) error {
```

Comment:
what if we want to update other than the name field or update the name and age together like that?
make update function generalize.

Reply:
After updating we should return updated data.
func (u *userStore) Update(id int, name string) (*models.User,error)